internal/supervisor/cluster_test.go

```diff
@@ -140,6 +149,61 @@ func TestClusterSupervisorDo(t *testing.T) {
 		<-mockStore.UnlockChan
 		require.Equal(t, 3, mockStore.UpdateClusterCalls)
 	})
+
+	t.Run("order of pending works", func(t *testing.T) {
+		logger := testlib.MakeLogger(t)
+		mockStore := &mockClusterStore{}
+
+		clusters := []*model.Cluster{
+			{
+				ID:    "1",
+				State: model.ClusterStateDeletionRequested, // should be 4th
+			},
+			{
+				ID:    "2",
+				State: model.ClusterStateCreationRequested, // should be 1st
+			},
+			{
+				ID:    "3",
+				State: model.ClusterStateResizeRequested, // should be 5th
+			},
+			{
+				ID:    "4",
+				State: model.ClusterStateCreationInProgress, // should be 3rd
+			},
+			{
+				ID:    "5",
+				State: model.ClusterStateProvisioningRequested, // should be 6th
+			},
+			{
+				ID:    "6",
+				State: model.ClusterStateCreationRequested, // should be 2nd
+			},
+		}
+		preferredClusterOrder := []string{"2", "6", "4", "1", "3", "5"}
+
+		mockStore.UnlockedClustersPendingWork = clusters
+		mockStore.Clusters = clusters
+
+		mockEventProducer := &mockEventProducer{}
+		supervisor := supervisor.NewClusterSupervisor(
+			mockStore,
+			&mockClusterProvisioner{},
+			&mockAWS{},
+			mockEventProducer,
+			"instanceID",
+			logger,
+		)
+		err := supervisor.Do()
+		require.NoError(t, err)
+
+		clusterListByWorkOrder := mockEventProducer.clusterListByEventOrder
+		require.Equal(t, len(preferredClusterOrder), len(clusterListByWorkOrder))
+		for i, clusterID := range clusterListByWorkOrder {
```

Comment:
I think you could use `Equal` directly to compare that the two slices and it's order are the same.

Reply:
Thanks. That will be nice